pkg/virt-controller/watch/replicaset_test.go

```diff
@@ -59,8 +66,7 @@ var _ = Describe("Replicaset", func() {
 			stop = make(chan struct{})
 			ctrl = gomock.NewController(GinkgoT())
 			virtClient := kubecli.NewMockKubevirtClient(ctrl)
```

Comment:
Can we get rid of `virtClient` completely?

Reply:
We cannot unfortunately. This is because the fakeclient does not implement the `KubevirtClient` interface.
The latter defines the "shortcut" methods for:
```
VirtualMachineInstance(namespace string) VirtualMachineInstanceInterface
VirtualMachineInstanceMigration(namespace string) VirtualMachineInstanceMigrationInterface
ReplicaSet(namespace string) ReplicaSetInterface
VirtualMachinePool(namespace string) poolv1.VirtualMachinePoolInterface
VirtualMachine(namespace string) VirtualMachineInterface
KubeVirt(namespace string) KubeVirtInterface
VirtualMachineInstancePreset(namespace string) VirtualMachineInstancePresetInterface
VirtualMachineSnapshot(namespace string) vmsnapshotv1alpha1.VirtualMachineSnapshotInterface
VirtualMachineSnapshotContent(namespace string) vmsnapshotv1alpha1.VirtualMachineSnapshotContentInterface
VirtualMachineRestore(namespace string) vmsnapshotv1alpha1.VirtualMachineRestoreInterface
VirtualMachineExport(namespace string) vmexportv1alpha1.VirtualMachineExportInterface
VirtualMachineInstancetype(namespace string) instancetypev1beta1.VirtualMachineInstancetypeInterface
VirtualMachineClusterInstancetype() instancetypev1beta1.VirtualMachineClusterInstancetypeInterface
VirtualMachinePreference(namespace string) instancetypev1beta1.VirtualMachinePreferenceInterface
VirtualMachineClusterPreference() instancetypev1beta1.VirtualMachineClusterPreferenceInterface
```
and others, and these methods are not generated by the fakeclient.
Passing the `fakeclient` to the controller creation causes a build error since the controller' `services.TemplateService` requires a `kubecli.KubevirtClient`.